src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,303 @@
+/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeApi = Interop.Diagnostics;
+
+namespace Tizen.System
+{
+    /// <summary>
+    /// Enumeration for the context types.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public enum ContextType
+    {
+        /// <summary>
+        /// Crash context generated by crash-worker.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        Crash = 0,
+    }
+
+    /// <summary>
+    /// Enumeration for the report types.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public enum ReportType
+    {
+        /// <summary>
+        /// Full crash report (.tar.gz).
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        CrashFull = 0,
+        /// <summary>
+        /// Crash report in JSON format.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        CrashInfoJson,
+    }
+
+    /// <summary>
+    /// The Diagnostics class is used for requesting client data and receiving diagnostics report events.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class Diagnostics : IDisposable
+    {
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Event to be invoked when new report context arrives.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<ReportReceivedEventArgs> ReportReceived;
+
+        /// <summary>
+        /// The Diagnostics class constructor.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen> 
+        public Diagnostics()
+        {
+            this.SetNotificationCallback();
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the Diagnostics class.
+        /// </summary>
+        ~Diagnostics()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Requests other apps to dump diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <feature>
+        /// http://tizen.org/privilege/internal/default/platform
+        /// </feature>
+        public DiagnosticsData RequestClientData(string clientId, string[] parameters)
```

Comment:
Parameters of methods must have valid documentation. Method has 2 parameters, 0 are documented. 
Method which returns non-void value must have exactly one `returns` section.